puppet.go

```diff
@@ -354,19 +354,19 @@ func (puppet *Puppet) updateAvatar(ctx context.Context, source *User, info *type
 			return true
 		}
 		avatarContentType = http.DetectContentType(avatarData)
+		puppet.AvatarPath = info.Profile.AvatarPath
```

Comment:
This was moved to before the `return true` so that an path change will end up getting saved to the store even when the hash didn't change.